sw/device/rom_exts/manifest.md

```diff
@@ -358,3 +366,77 @@ B, without having to re-link or re-compile the image (note we only have one
 entry address). At least initially, we will only create images that can work
 from Slot A, and later we will add support ensuring these images are linked in
 a way that they can run from Slot B without modification.
+
+
+## How To Validate a ROM_EXT Image
+
+Below we provide an abstract definition of how to sign and verify a ROM_EXT
+image. This description is provided so that there is a specification of what the
+signing and validation systems are implementing, so that alternate
+implementations may be produced if necessary.
+
+Notation:
+
+*   `rom_ext` - ROM_EXT Image, including manifest and contents (Variable Length)
+    `signed_area(rom_ext)` denotes just the area between the start of the signed
+    contents and the end of the image.
+*   `device_usage_value` - Device-Calculated Usage Constraint Value (1024
+    bytes). This is the "device usage value" described above, as calculated by
+    the Mask ROM.
+*   `system_state_value` - Device System State Value (32 bytes). This is a Mask
+    ROM calculated value that summarises the current device state.
+*   `private_key` - RSA Private Key
+*   `public_key` - RSA Public Key
+*   `signature` - ROM_EXT RSA Signature,
+*   `||` - Denotes concatenation without a delimiter.
+*   `:=` - Denotes assignment.
+
+### Algorithms
+
+*   We primarily use
+    [`RSASSA-PKCS1-V1_5-SIGN`](https://tools.ietf.org/html/rfc3447#section-8.2.1)
+    and
+    [`RSASSA-PKCS1-V1_5-VERIFY`](https://tools.ietf.org/html/rfc3447#section-8.2.2)
+    for signatures and verification.
+*   RSA Signing and Verification is done with 3072-bit keys.
+*   The `Hash` Operation may be done with SHA2-265, SHA2-384, or SHA2-512,
+    depending on the Signature Algorithm Identifier provided in the manifest.
```

Comment:
`EMSA-PKCS1-V1_5-ENCODE` (first step of `RSASSA-PKCS1-V1_5-SIGN`) already [encodes](https://tools.ietf.org/html/rfc3447#section-9.2) the hash algorithm in `DigestInfo`. If this field is *only* for identifying the hash algorithm, IMHO it is redundant. I'm not sure if we are planning to use other signature schemes though, e.g. `RSASSA-PKCS1-v1_5` vs `RSASSA-PSS`.

Reply:
Garret and I discussed this, and the answer is you're probably right - We'll make the "ROM_EXT Manifest Identifier" identify a list of allowed Hash implementations, and that should solve what we need. I'm not sure how we ensure that the Signature Verification module is giving us a buffer back of the right size to match the hash, but we can work that out later.

Reply:
ack. We should track this in an issue to finalize the selection of a hash function.  We can compare performance, digest size and security level as part of this. 

Focused mainly on security level, I believe it is okay to choose `SHA2-256` or `SHA3-256`.

We should be able to fix the digest sizes as part of this discussion.